File: Codes/Kan Zheng/C++/Hashtable/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> vec=nums;
        vector<int> result;
        int flag=0;
        for(int i=0;i<vec.size();i++)
        {
            for(int j=i+1;j<vec.size();j++)
            {
                if(vec[i]+vec[j]==target)
                {
                     result.push_back(i+1);  
                     result.push_back(j+1);
                     flag=1;
                     break;
                }
            }
            if(flag==1)
            break;
        }
        return result;
    }
};

File: Codes/Bingran Li/HashTable/1. Two Sum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        // the value is the index of key
        HashMap<Integer, Integer> myMap = new HashMap<Integer,Integer>();
        int []result = new int[2];
        
        for(int i=0;i<nums.length;i++){
            if(myMap.containsKey(target-nums[i])){
                result[1]=i+1;
                result[0]=myMap.get(target-nums[i]);
                return result; // miss return, waste time
            }
            else
                myMap.put(nums[i],i+1);
        }
        return result;
    }
}